src/client.cc

```diff
@@ -117,39 +118,40 @@ DisplayCoord Client::dimensions() const
     return m_ui->dimensions();
 }
 
+String generate_context_info(const Context& context)
+{
+    String s = "";
+    if (context.buffer().is_modified())
+        s += "[+]";
+    if (context.client().input_handler().is_recording())
+        s += format("[recording ({})]", context.client().input_handler().recording_reg());
+    if (context.buffer().flags() & Buffer::Flags::New)
+        s += "[new file]";
+    if (context.hooks_disabled())
+        s += "[no-hooks]";
+    if (context.buffer().flags() & Buffer::Flags::Fifo)
+        s += "[fifo]";
+    return s;
```

Comment:
I have just realized I don't know C++. Where is this string allocated? When will it be deleted? Is it safe to return it like this? It is sent to `parse_display_line` below.